### PR TITLE
Mobx trainer

### PR DESCRIPTION
**Pull request by soyarsauce**
### What this PR does

Fixes https://github.com/TerriaJS/terriajs/issues/4421

Doesn't include the select styles as we are gauging whether to implement them or not

### Checklist

-   [x] n/a unit tests until confirming implementation
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

**Comment by soyarsauce** [comment]
test results
image

**Comment by soyarsauce** [comment]
@steve9164 let me know how you go with this one

**Review by tephenavies — COMMENTED** [review]
This all looks good and works well.

File: lib/ReactViews/Map/TrainerBar/TrainerBar.tsx [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+              />
+            )}
+            onChange={(e: React.ChangeEvent<HTMLSelectElement>) =>
+              viewState.setCurrentTrainerItemIndex(Number(e.target.value))
+            }
+            value={viewState.currentTrainerItemIndex}
+            // css={"min-width: 280px;"}
+          >
+            {selectedTrainerItems.map((item, index) => (
+              <option value={index}>{item.title}</option>
```
Comment (zoran995) [inline lib/ReactViews/Map/TrainerBar/TrainerBar.tsx]: this is probably missing a key

Comment (soyarsauce) [reply lib/ReactViews/Map/TrainerBar/TrainerBar.tsx]: Thanks good pick up - added

File: lib/ReactViews/Map/MenuBar.jsx [hunk 2]
```diff
@@ -168,6 +178,7 @@ MenuBar.propTypes = {
   animationDuration: PropTypes.number,
   menuItems: PropTypes.arrayOf(PropTypes.element),
   menuLeftItems: PropTypes.arrayOf(PropTypes.element),
+  theme: PropTypes.object.isRequired,
```
[the remark this answers is not in the document]
Comment (soyarsauce) [reply lib/ReactViews/Map/MenuBar.jsx]: I think it was an older attempt of adding the theme, now it just uses from css theme via provider

**Comment by soyarsauce** [comment]
@steve9164 please approve if all good now

**Review by tephenavies — APPROVED** [review]

**Pull request merged**